src/client/Microsoft.Identity.Client/ManagedIdentity/ManagedIdentityClient.cs

```diff
@@ -29,6 +34,17 @@ public ManagedIdentityClient(RequestContext requestContext)
             }
         }
 
+        /// <summary>
+        /// Resets the cached managed identity source. Used only for testing purposes.
+        /// </summary>
+        internal static void ResetManagedIdentitySourceCache()
```

Comment:
I think this will need to be made public for Azure SDK to test. And it'd be better to have a "ResetStaticCaches" general API, like our tests currently do, which reset everything.

Reply:
i.e. see the logic in TestCommon.ResetInternalStaticCaches();